### PR TITLE
EO-713 Signals Mailbox Provider Facet

### PR DESCRIPTION
**Pull request by 21042jim**
EO-713 Signals Mailbox Provider Facet

### What Changed
 - Added facets property to FacetFilter component
 - Removed mb provider facet on v1 pages and v2 spam hits pages and default to "No Breakdown"

### How To Test
- [Point your local upstreams to staging or production](https://confluence.int.messagesystems.com/display/ENG/Configuring+Openresty+Upstreams)
- Log into appteam or account 108 for data

### Things to Verify
- Where to verify:
  - All details pages
  - All V1 summary tables
  - All V2 dashboards (with summary tables)
- What to verify:
  - Facet selection
  - Tables and Charts going back and forth from spam hits pages

### To Do
- [ ] Address any feedback

### CODE REVIEW

File: src/pages/signals/dashboards/SpamTrapDashboard.js [hunk 1]
```diff
@@ -24,10 +26,10 @@ export class SpamTrapDashboard extends Component {
               <DateFilter />
             </Grid.Column>
             <SubaccountFilter />
-            <FacetFilter />
+            <FacetFilter facets={_.reject(facets, (facet) => facet.key === 'mb_provider')}/>
           </Grid>
         </Panel>
-        <SpamTrapOverview subaccounts={subaccounts} hideTitle />
+        <SpamTrapOverview subaccounts={subaccounts} hideTitle facets={_.reject(facets, (facet) => facet.key === 'mb_provider')}/>
```
Comment (jonambas) [inline src/pages/signals/dashboards/SpamTrapDashboard.js]: I don't think you need `facets` here, it isn't used in SpamTrapOverview

File: src/pages/signals/components/SpamTrapOverview.js [hunk 2]
```diff
@@ -52,6 +52,9 @@ class SpamTrapOverview extends React.Component {
 
   getData = () => {
     const { getSpamHits, signalOptions, summaryTable } = this.props;
+    if (signalOptions.facet === 'mb_provider') {
+      signalOptions.facet = '';
+    }
```
Comment (jonambas) [inline src/pages/signals/components/SpamTrapOverview.js]: will need to also clear `signalOptions.facetSearchTerm`

**Comment by 21042jim** [comment]
The redux form fields and the downshift items are so tightly coupled, I didn't see a place to convert the friendly form names back to ids unless i put something in the action perhaps? also, ip pools also uses this but has no friendly names to convert.

**Comment by 21042jim** [comment]
the titles are strictly facet values, so for ip pools it currently says "shared" for Master Account. seems like that could be another ticket if we want to change that.

**Comment by jonambas** [comment]
Yeah okay, ip pool naming is weird too. I'll open another ticket. We could transform them in the tables and titles directly though instead of going through redux.

**Pull request merged**